docs/content/sips/00x-cloudevent-trigger.md

```diff
@@ -0,0 +1,263 @@
+}
+```
+
+Creating an HTTP CloudEvents trigger is done by defining the top level application trigger in spin. The following code snippet shows the definition of a HTTP CloudEvents trigger.
+```toml
+# spin.toml
+trigger = { type = "http", base = "/", schema = "cloudevents" }
+```
+
+The added `schema` attribute in trigger will tell spin application that it will expect the HTTP request and responses are CloudEvents.
+
+> Note that the `schema` attribute is not the same as the `schema` attribute in the CloudEvents spec. The `schema` attribute in the spec is the schema of the payload.
+
+We also allow users to define individual component to be CloudEvents components. For example, we could define a HTTP CloudEvents component in spin.toml:
+
+```toml
+# spin.toml
+trigger = { type = "http", base = "/" }
+
+[[component]]
+id = "hello"
+source = "target/wasm32-wasi/release/spinhelloworld.wasm"
+description = "A simple component that returns hello."
+schema = "cloudevents"
+[component.trigger]
+route = "/hello"
+```
+
+> Note that if there is no `schema` attribute in the application or individual component, the HTTP request and responses are not CloudEvents.
+
+## The WebAssembly interface
+
+The CloudEvent trigger is built on top of the
+[WebAssembly component model](https://github.com/WebAssembly/component-model).
+The current interface is defined using the
+[WebAssembly Interface (WIT)](https://github.com/bytecodealliance/wit-bindgen/blob/main/WIT.md)
+format, and is a function that takes the event payload as its only parameter:
+
+```fsharp
+// wit/ephemeral/spin-ce.wit
+
+// The event payload.
+record event {
+    // The event type.
+    type: string,
+    // The event id.
+    id: string,
+    // The event source.
+    source: string,
+    // The event specversion.
+    specversion: string,
+    // The event data content type.
+    datacontenttype: string,
+    // The event data schema.
+    dataschema: string,
+    // The event subject.
+    subject: string,
+    // The event time.
+    time: option<string>,
+    // The event data.
+    data: option<string>
+}
+
+// The entry point for a CloudEvent handler.
+handle-cloudevent: function(event: event) -> expected<event, error>
+```
+
+
+This is the function that all CloudEvents components must implement, and which is
+used by the Spin Redis executor when instantiating and invoking the component.
+
+Notice that the function will return a `expected<event, error>` value. If the sink address is not set, the function will return `expected<event, error>` with `event` as the value. If the sink address is set, spin will make an outbound HTTP request to the sink address and return the response as the value.
+
+Due to the [known issue](https://github.com/bytecodealliance/wit-bindgen/issues/171) of the cannonical ABI representation cannot exceeds the number of parameters (16) in the wasmtime API, the proposed WIT format cannot be compiled. I am proposing an alternative WIT format for CloudEvents:
```

Comment:
I think this is relatively close to being solved — https://github.com/bytecodealliance/wit-bindgen/issues/201

Reply:
We need to update our dependency on wit-bindgen, with a new rev commit number